argo.tf

```diff
@@ -1,4 +1,36 @@
 locals {
+  crds_argo_application_metadata = {
+    "labels" : try(var.crds_argo_metadata.labels, {}),
+    "annotations" : try(var.crds_argo_metadata.annotations, {}),
+    "finalizers" : try(var.crds_argo_metadata.finalizers, [])
+  }
+  crds_argo_application_values = {
+    "project" : var.argo_project
+    "source" : {
+      "repoURL" : var.helm_repo_oci ? local.helm_repo_url : "https://${local.helm_repo_url}"
+      "chart" : var.crds_helm_chart_name
+      "targetRevision" : var.helm_chart_version
+      "helm" : {
+        "releaseName" : "${var.helm_release_name}-crds"
+        "parameters" : [for k, v in var.crds_settings : tomap({ "forceString" : true, "name" : k, "value" : v })]
+        "values" : var.enabled ? data.utils_deep_merge_yaml.crds_values[0].output : ""
+      }
+    }
+    "destination" : {
+      "server" : var.argo_destination_server
+      "namespace" : var.namespace
+    }
+    "ignoreDifferences" : [{
+      "group" : "apiextensions.k8s.io"
+      "kind" : "CustomResourceDefinition"
+      "jqPathExpressions" : [
+        ".spec.names.shortNames"
+      ]
+    }]
```

Comment:
What is the reason to ignore diffs for the `spec.names.shortNames` ?

Reply:
This is a copy-paste addition from Linkerd CRDs. Not needed for Karpenter.